File: 2021 czerwiec/c++/4.cpp
#include <iostream>
#include <fstream>
#include <string>
using namespace std;


int main()
{
    ifstream in("napisy.txt");
    
    string s, ans;
    while (in >> s)
    {
        string tmp;
        if (ans.size() < 3 || ans.substr(ans.size() - 3, 3) != "XXX")
        {
            for (char c: s)
            {
                if (c >= '0' && c <= '9')
                {
                    tmp.push_back(c);
                    if (tmp.size() == 2)
                    {
                        int x = stoi(tmp);
                        if (x >= 65 && x <= 90)
                            ans.push_back(x);
                        tmp.clear();
                    }
                }
            }
        }
    }
    cout << ans << '\n';
}

File: 2021 czerwiec/python/4.py
with open("napisy.txt", "r") as f:
    file = f.read().strip().split('\n')

ans = ""
for s in file:
    if ans[-3:] == "XXX":
        break

    tmp = ""
    for i in s:
        if i in '0123456789':
            tmp += i
            if len(tmp) == 2:
                if int(tmp) >= 65 and int(tmp) <= 90:
                    ans += chr(int(tmp))
                tmp = ""

print(ans) 